### DISCUSSION TITLE
macOS "Encryption is insufficient."

### DISCUSSION THREAD

**Discussion by dmkrueger** [gopro/OpenGoPro#149] [Q&A]
Hello, just trying to run the GoPro-wifi python example. I get "Encryption is insufficient." when trying to connect to the camera over BLE. I am running python3.9.10

Terminal trace back
```
gopro-wifi             
INFO     20:07:08.703 Establishing the BLE connection                                                                                                                         client.py:102
ERROR    20:07:09.086 Error while opening: Failed to read descriptor 50: Error Domain=CBATTErrorDomain Code=15 "Encryption is insufficient."                                     gopro.py:374
         UserInfo={NSLocalizedDescription=Encryption is insufficient.}                                                                                                                       
INFO     20:07:09.091 Terminating the BLE connection                        

```
Has anyone had this issue? I am running the latest release of macOS 15.4.

I have had no luck with this OpenGoPro package.

Thanks
Dean

**Comment by tcamise-gpsw** [comment]
Hello. This is usually because you have already paired to the camera but have deleted the device from your computer.

Please try resetting both sides as described [here](https://gopro.github.io/OpenGoPro/tutorials/python/connect-ble#device-not-connecting).

Also note that, depending on the MAcOS version, you may need to accept the pairing pop-up when it occurs.
